### ISSUE TITLE
assertions_on_constants raised on non constant expression

### ISSUE THREAD

**Issue by gwenn** [rust-lang/rust-clippy#16242]
### Summary

Rusqlite build fails https://github.com/rusqlite/rusqlite/actions/runs/20227751012/job/58063065463
```
107 | /         assert!(!(cfg!(feature = "bundled-windows") && !cfg!(feature = "bundled") && !win_target()),
108 | |             "This module should not be used: we're not on Windows and the bundled feature has not been enabled");
    | |________________________________________________________________________________________________________________^
    |
    = help: consider moving this into a const block: `const { assert!(..) }`
```
https://github.com/rusqlite/rusqlite/blob/78d66787959bc22d140902c4163d456848460554/libsqlite3-sys/build.rs#L107-L108
but `win_target` is not `const`:
https://github.com/rusqlite/rusqlite/blob/78d66787959bc22d140902c4163d456848460554/libsqlite3-sys/build.rs#L13-L15

### Lint Name

assertions_on_constants

### Reproducer

I tried this code:

```rust
fn win_target() -> bool {
    env::var("CARGO_CFG_WINDOWS").is_ok()
}

        assert!(!(cfg!(feature = "bundled-windows") && !cfg!(feature = "bundled") && !win_target()),
            "This module should not be used: we're not on Windows and the bundled feature has not been enabled");
```

I saw this happen:

```
warning: this assertion has a constant value
   --> libsqlite3-sys\build.rs:107:9
    |
107 | /         assert!(!(cfg!(feature = "bundled-windows") && !cfg!(feature = "bundled") && !win_target()),
108 | |             "This module should not be used: we're not on Windows and the bundled feature has not been enabled");
    | |________________________________________________________________________________________________________________^
```

I expected to see this happen:
No warning when expression is not a constant

### Version

```text
rustc 1.92.0 (ded5c06cf 2025-12-08) (Rev1, Built by MSYS2 project)
binary: rustc
commit-hash: ded5c06cf21d2b93bffd5d884aa6e96934ee4234
commit-date: 2025-12-08
host: x86_64-pc-windows-gnu
release: 1.92.0
LLVM version: 21.1.7
```
